File: lib/models/coffee_collection.dart
class CoffeeCollection {
  final String id;
  final String memberId;
  final String memberNumber;
  final String memberName;
  final String seasonId;
  final String seasonName;
  final String productType; // 'CHERRY' or 'MBUNI'
  final double grossWeight;
  final double tareWeight;
  final double netWeight;
  final int numberOfBags; // Number of bags used in the collection
  final DateTime collectionDate;
  final bool isManualEntry;
  final String? receiptNumber;
  final String? userId;
  final String? userName;
  final double? pricePerKg;
  final double? totalValue;

  CoffeeCollection({
    required this.id,
    required this.memberId,
    required this.memberNumber,
    required this.memberName,
    required this.seasonId,
    required this.seasonName,
    required this.productType,
    required this.grossWeight,
    this.tareWeight = 0.0,
    required this.netWeight,
    this.numberOfBags = 1,
    required this.collectionDate,
    required this.isManualEntry,
    this.receiptNumber,
    this.userId,
    this.userName,
    this.pricePerKg,
    this.totalValue,
  });

  factory CoffeeCollection.fromJson(Map<String, dynamic> json) {
    return CoffeeCollection(
      id: json['id'],
      memberId: json['memberId'],
      memberNumber: json['memberNumber'],
      memberName: json['memberName'],
      seasonId: json['seasonId'],
      seasonName: json['seasonName'],
      productType: json['productType'],
      grossWeight: json['grossWeight'].toDouble(),
      tareWeight: json['tareWeight']?.toDouble() ?? 0.0,
      netWeight: json['netWeight'].toDouble(),
      numberOfBags: json['numberOfBags']?.toInt() ?? 1,
      collectionDate: DateTime.parse(json['collectionDate']),
      isManualEntry: json['isManualEntry'] == 1,
      receiptNumber: json['receiptNumber'],
      userId: json['userId'],
      userName: json['userName'],
      pricePerKg: json['pricePerKg']?.toDouble(),
      totalValue: json['totalValue']?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'memberId': memberId,
      'memberNumber': memberNumber,
      'memberName': memberName,
      'seasonId': seasonId,
      'seasonName': seasonName,
      'productType': productType,
      'grossWeight': grossWeight,
      'tareWeight': tareWeight,
      'netWeight': netWeight,
      'numberOfBags': numberOfBags,
      'collectionDate': collectionDate.toIso8601String(),
      'isManualEntry': isManualEntry ? 1 : 0,
      'receiptNumber': receiptNumber,
      'userId': userId,
      'userName': userName,
      'pricePerKg': pricePerKg,
      'totalValue': totalValue,
    };
  }
  
  CoffeeCollection copyWith({
    String? id,
    String? memberId,
    String? memberNumber,
    String? memberName,
    String? seasonId,
    String? seasonName,
    String? productType,
    double? grossWeight,
    double? tareWeight,
    double? netWeight,
    int? numberOfBags,
    DateTime? collectionDate,
    bool? isManualEntry,
    String? receiptNumber,
    String? userId,
    String? userName,
    double? pricePerKg,
    double? totalValue,
  }) {
    return CoffeeCollection(
      id: id ?? this.id,
      memberId: memberId ?? this.memberId,
      memberNumber: memberNumber ?? this.memberNumber,
      memberName: memberName ?? this.memberName,
      seasonId: seasonId ?? this.seasonId,
      seasonName: seasonName ?? this.seasonName,
      productType: productType ?? this.productType,
      grossWeight: grossWeight ?? this.grossWeight,
      tareWeight: tareWeight ?? this.tareWeight,
      netWeight: netWeight ?? this.netWeight,
      numberOfBags: numberOfBags ?? this.numberOfBags,
      collectionDate: collectionDate ?? this.collectionDate,
      isManualEntry: isManualEntry ?? this.isManualEntry,
      receiptNumber: receiptNumber ?? this.receiptNumber,
      userId: userId ?? this.userId,
      userName: userName ?? this.userName,
      pricePerKg: pricePerKg ?? this.pricePerKg,
      totalValue: totalValue ?? this.totalValue,
    );
  }
} 

File: InukaAdmin/src/main/java/com/codejar/inukaadmin/model/CoffeeCollection.java
package com.codejar.inukaadmin.model;

import java.util.Date;

public class CoffeeCollection {
    private String id;
    private String memberId;
    private String memberNumber;
    private String memberName;
    private Date collectionDate;
    private String seasonId;
    private String seasonName;
    private String productType;
    private double grossWeight;
    private double tareWeight;
    private double netWeight;
    private int numberOfBags;
    private double pricePerKg;
    private double totalValue;
    private String receiptNumber;
    private boolean isManualEntry;
    private String collectedBy;
    private String userId;
    private Date createdAt;
    
    // Constructors
    public CoffeeCollection() {}
    
    // Getters and setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    
    public String getMemberId() { return memberId; }
    public void setMemberId(String memberId) { this.memberId = memberId; }
    
    public String getMemberNumber() { return memberNumber; }
    public void setMemberNumber(String memberNumber) { this.memberNumber = memberNumber; }
    
    public String getMemberName() { return memberName; }
    public void setMemberName(String memberName) { this.memberName = memberName; }
    
    public Date getCollectionDate() { return collectionDate; }
    public void setCollectionDate(Date collectionDate) { this.collectionDate = collectionDate; }
    
    public String getSeasonId() { return seasonId; }
    public void setSeasonId(String seasonId) { this.seasonId = seasonId; }
    
    public String getSeasonName() { return seasonName; }
    public void setSeasonName(String seasonName) { this.seasonName = seasonName; }
    
    public String getProductType() { return productType; }
    public void setProductType(String productType) { this.productType = productType; }
    
    public double getGrossWeight() { return grossWeight; }
    public void setGrossWeight(double grossWeight) { this.grossWeight = grossWeight; }
    
    public double getTareWeight() { return tareWeight; }
    public void setTareWeight(double tareWeight) { this.tareWeight = tareWeight; }
    
    public double getNetWeight() { return netWeight; }
    public void setNetWeight(double netWeight) { this.netWeight = netWeight; }
    
    public int getNumberOfBags() { return numberOfBags; }
    public void setNumberOfBags(int numberOfBags) { this.numberOfBags = numberOfBags; }
    
    public double getPricePerKg() { return pricePerKg; }
    public void setPricePerKg(double pricePerKg) { this.pricePerKg = pricePerKg; }
    
    public double getTotalValue() { return totalValue; }
    public void setTotalValue(double totalValue) { this.totalValue = totalValue; }
    
    public String getReceiptNumber() { return receiptNumber; }
    public void setReceiptNumber(String receiptNumber) { this.receiptNumber = receiptNumber; }
    
    public boolean isManualEntry() { return isManualEntry; }
    public void setManualEntry(boolean manualEntry) { isManualEntry = manualEntry; }
    
    public String getCollectedBy() { return collectedBy; }
    public void setCollectedBy(String collectedBy) { this.collectedBy = collectedBy; }
    
    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }
    
    public Date getCreatedAt() { return createdAt; }
    public void setCreatedAt(Date createdAt) { this.createdAt = createdAt; }
}
